libgnucash/app-utils/gnc-gsettings.c

```diff
@@ -850,3 +918,25 @@ void gnc_gsettings_version_upgrade (void)
     if (cur_maj_min > old_maj_min)
         gnc_gsettings_set_int (GNC_PREFS_GROUP_GENERAL, GNC_PREF_VERSION, cur_maj_min);
 }
+
+
+void gnc_gsettings_block_all (void)
+{
+    PINFO("registered_handlers_hash block list size before is %d",
+           g_hash_table_size (registered_handlers_hash));
+    g_hash_table_foreach_remove (registered_handlers_hash,
+                                 handlers_hash_helper, GINT_TO_POINTER(BLOCK));
```

Comment:
As I said in earlier comments it doesn't make sense to combine blocking signals with a hash table cleanup (that is removing stale handler ids). So g_hash_table_foreach_remove is out of place here.
Instead you can rewrite the helper function to block or unblock the passed in handlerid based on a boolean passed in (instead of the enum). In fact as the hash helper function is so minimal, I would write two: one for blocking a signal handler and one for unblocking it. And then don't pass anything as user data. But that's a free choice.

If you worry this helper function could ever get called on a handler that's no longer connected you can add a test in the helper first which emits a warning and returns if you run into this situation. I expect the code to never hit it though.